src/main/java/hlf/java/rest/client/service/impl/EventPublishServiceImpl.java

```diff
@@ -82,18 +84,51 @@ public void onFailure(Throwable ex) {
   }
 
   @Override
-  public boolean publishChaincodeEvents(
+  public void publishChaincodeEvents(
       String payload,
       String chaincodeName,
       String fabricTxId,
       String eventName,
       String channelName,
       String messageKey) {
-    boolean status = true;
 
+    Optional<FabricProperties.ChaincodeDetails> optionalChaincodeDetails =
+        fabricProperties.getEvents().getChaincodeDetails().stream()
+            .filter(
+                chaincodeDetail ->
+                    chaincodeDetail.getChannelName().equals(channelName)
+                        && chaincodeDetail.getChaincodeId().equals(chaincodeName))
+            .findAny();
+
+    if (!optionalChaincodeDetails.isPresent()
+        || CollectionUtils.isEmpty(optionalChaincodeDetails.get().getListenerTopics())) {
+      sendMessage(
+          kafkaProperties.getEventListener().getTopic(),
+          payload,
+          chaincodeName,
+          fabricTxId,
+          eventName,
+          channelName,
+          messageKey);
+      return;
+    }
+
+    for (String topic : optionalChaincodeDetails.get().getListenerTopics()) {
+      sendMessage(topic, payload, chaincodeName, fabricTxId, eventName, channelName, messageKey);
```

Comment:
Does all these topics rely on the same Kafka Template? What if they belong to different outbound brokers?

Reply:
All topics rely on the same kafka template. We can provide multiple brokers in `brokerHost` field in application.yml in comma separated manner

Reply:
updated code to create multiple producers and using `RoutingKafkaTemplate` to dispatch message based on topic name configured while creating producer factory